feat: technical reference

mfw78 opened the pull request:
# Description
This PR contains the canonical Technical Reference for CoW Protocol.

# Changes

- [x] Implemented core / periphery methodology 
- [x] Migrated previous documentation, and had proof-read
- [x] All contracts extensively documented
- [x] Typedoc generation for SDKs
- [x] Interactive swagger use for APIs

## Related Issues

Fixes #6, #8

mfw78 committed:
feat: technical reference

mfw78 committed:
fix: build issues

mfw78 committed:
fix: bump cow-sdk

mfw78 committed:
chore(reference): remove sub-category indexes

mfw78 committed:
chore(reference): consolidate appData

mfw78 committed:
chore: add json schema plugin

mfw78 committed:
docs(cow-protocol): contracts technical reference

mfw78 committed:
chore: typedoc appdata

mfw78 committed:
chore: docs at root

mfw78 committed:
docs(reference): app data

MartinquaXD reviewed (commented):
Really good stuff! 👏 
Only nits but didn't get through everything, yet.

fedgiac reviewed (commented):
Partial review of the contracts section, core + ETH-flow. The core is basically spotless, really nice!

fedgiac reviewed (commented):
I finished my review of the contract section. My understanding of ConditionalCoW is still a bit shallow, so I'm not sure all comments actually make sense.

fhenneke reviewed (commented):
Went through the rewards and slippage part of the docs. @harisang could you have a quick look as well?

mfw78 committed:
Apply suggestions from code review

Co-authored-by: Federico Giacon <58218759+fedgiac@users.noreply.github.com>
Co-authored-by: Felix Henneke <felix.henneke@protonmail.com>

mfw78 committed:
chore: review comments

mfw78 commented:
OK, I've addressed most of the above comments / integrated suggestions. I'm merging this, leaving the above unresolved, to come back to towards the top of the waterfall / CoW DAO-wide review of docs.
